### PR TITLE
Add support for Configuration Slicing plugin

### PR DESCRIPTION
**Pull request by mgaunin**
Really useful, easy to configure, easy to use

### CODE REVIEW

**Comment by mgaunin** [comment]
@reviewbybees please

**Comment by aheritier** [comment]
LGTM 🐝

File: src/main/java/org/jenkinsci/plugins/emailext_template/configurationslicing/ExtEmailTemplateSlicer.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+import hudson.Extension;
+import hudson.model.AbstractProject;
+import hudson.tasks.Publisher;
+import org.jenkinsci.plugins.emailext_template.ExtendedEmailPublisherTemplate;
+import org.jenkinsci.plugins.emailext_template.ExtendedEmailTemplatePublisher;
+import org.jenkinsci.plugins.emailext_template.Messages;
+import org.jenkinsci.plugins.emailext_template.TemplateId;
+
+import java.io.IOException;
+import java.util.*;
```
Comment (amuniz) [inline src/main/java/org/jenkinsci/plugins/emailext_template/configurationslicing/ExtEmailTemplateSlicer.java]: Don't use wildcard imports please.

**Comment by mgaunin** [comment]
@amuniz I did not notice that my IDE replaced imports with wildcard one, sorry for that

File: src/main/java/org/jenkinsci/plugins/emailext_template/configurationslicing/ExtEmailTemplateSlicer.java (hunk 1 above)
Comment (mgaunin) [inline src/main/java/org/jenkinsci/plugins/emailext_template/configurationslicing/ExtEmailTemplateSlicer.java]: fixed

**Comment by amuniz** [comment]
Looks good :bee:

**Comment by mgaunin** [comment]
@slide any chance to merge and release soon? Thanks

File: src/main/java/org/jenkinsci/plugins/emailext_template/ExtendedEmailTemplatePublisher.java [hunk 2]
```diff
@@ -51,7 +57,7 @@ public boolean prebuild(AbstractBuild<?, ?> build, BuildListener listener) {
             if(t != null) {
                 result &= t.getPublisher().prebuild(build, listener);
             } else {
-                listener.getLogger().println(Messages.ExtendedEmailTemplatePublisher_TemplateRemoved(template.getTemplateId()));
+                listener.getLogger().println(Messages.TemplateIdRemoved(template.getTemplateId()));
```
Comment (slide) [inline src/main/java/org/jenkinsci/plugins/emailext_template/ExtendedEmailTemplatePublisher.java]: Why change this?

File: src/test/java/org/jeninsci/plugins/emailext_template/ExtendedEmailTemplatePublisherTest.java [hunk 3]
```diff
@@ -1,59 +1,59 @@
...
-                !b.getLog(100).contains(Messages.ExtendedEmailTemplatePublisher_TemplateRemoved(template.getId())));
-
-        descriptor.removeTemplateById(template.getId());
-
-        b = (AbstractBuild)p.scheduleBuild2(0, new Cause.UserCause()).get();
-        assertTrue("Template was removed, so we should get the message",
-                b.getLog(100).contains(Messages.ExtendedEmailTemplatePublisher_TemplateRemoved(template.getId())));
-    }
-}
+package org.jeninsci.plugins.emailext_template;
```
Comment (slide) [inline src/test/java/org/jeninsci/plugins/emailext_template/ExtendedEmailTemplatePublisherTest.java]: Why was this changed?

File: src/main/java/org/jenkinsci/plugins/emailext_template/ExtendedEmailTemplatePublisher.java (hunk 2 above)
Comment (mgaunin) [inline src/main/java/org/jenkinsci/plugins/emailext_template/ExtendedEmailTemplatePublisher.java]: Reformulated the message to prevent duplication (needed it in another class)

File: src/test/java/org/jeninsci/plugins/emailext_template/ExtendedEmailTemplatePublisherTest.java (hunk 3 above)
Comment (mgaunin) [inline src/test/java/org/jeninsci/plugins/emailext_template/ExtendedEmailTemplatePublisherTest.java]: IntelliJ stuff... look at the real change in this class so same than the other comment: Reformulated the message to prevent duplication (needed it in another class)
